pkg/operator/ceph/cluster/mgr/mgr.go

```diff
@@ -339,9 +340,57 @@ func startModuleConfiguration(description string, configureModules func() error)
 }
 
 // Ceph docs about the prometheus module: http://docs.ceph.com/docs/master/mgr/prometheus/
-func (c *Cluster) enablePrometheusModule() error {
-	if err := cephclient.MgrEnableModule(c.context, c.clusterInfo, PrometheusModuleName, true); err != nil {
-		return errors.Wrap(err, "failed to enable mgr prometheus module")
+func (c *Cluster) configurePrometheusModule() error {
+	if !c.spec.Prometheus.Disabled {
```

Comment:
We can use this setting that already exists
```suggestion
	if c.spec.Monitoring.Enabled {
```

Reply:
`bool` type default value is false,  `Enabled` will cause old cluster internal mgr prometheus module disabled after update